### ISSUE TITLE
Lint prefer_final_fields fails for private field modified outside of containing class

### ISSUE THREAD

**Issue by donny-dont** [dart-lang/linter#325]
I have a false positive from `prefer_final_fields` when a private field is modified outside of the containing class. This can happen in tightly coupled classes.

```
class Foo {
  Bar _value;

  Bar get value => _value;
}

class Bar {
  void linkFoo(Bar value) {
    assert(value._value == null);
    value._value = this;
  }
}
```

[Real life example](https://github.com/dogma-dart/dogma-source-analyzer/blob/master/lib/src/metadata/enclosing_metadata.dart)

**Comment by alexeieleusis** [comment]
This is certainly an omission in the rule, I guess I assumed one class per file. I personally would create a private setter on `Foo` and use that from `Bar` and solves your problem while the fix for this bugs lands.

**Comment by donny-dont** [comment]
I'm fine with waiting for a fix @alexeieleusis. I'm not a huge fan of tightly coupled classes like this but it ensures correctness in this case. Thanks for taking a look into it!
